pkg/operator/configmetrics/configmetrics.go

```diff
@@ -59,6 +59,22 @@ func (m *configMetrics) Collect(ch chan<- prometheus.Metric) {
 			}
 			g.Set(value)
 			ch <- g
+		} else {
+			// if the parent has not reported platform status yet
+			infraType := infra.Status.Platform
+			var g prometheus.Gauge
+			var value float64 = 1
+			switch {
+			// it is illegal to set type to empty string, so let the default case handle
+			// empty string (so we can detect it) while preserving the constant None here
```

Comment:
I'm confused by this. So if we find `None`, we set the value to `0`. Otherwise, it's set to `1` (including the empty string). Why bother setting the value to `0` at all? We can already distinguish between <empty> and `None`.

Reply:
The query we'd want to do most often is "how many non-cloud provider clusters do we have" and "how many cloud-provider clusters do we have" so 1 vs 0 is that check.  The error state is not the same as having no cloud provider, so I'd rather it be treated as a bug when doing the naive queries (the query is `count by (_id) (cloud_provider == 1)` or `count by (_id) (cloud_provider == 0)`.  I didn't want someone reading this to assume "" == "None" because it is not.